js/linked-list.js

```diff
@@ -0,0 +1,99 @@
+"use strict";
+
+class Node {
+    constructor(data, next=null){
+        this.data = data;
+        this.next = next;
+    }
+}
+
+class LinkedList {
+    constructor(iterable=null) {
+        this.head = null;
+        this._counter = 0;
+        if(Array.isArray(iterable)){
+            iterable.map(x => this.push(x));
+        }
+    }
+
+    push(val){
+        let newHead = new Node(val, this.head);
+        this.head = newHead;
+        this._counter ++;
+    }
+
+    pop(){
+        if(this.head === null){
+            throw new Error('IndexError: Nothing to pop.');
```

Comment:
Errors! Javascript has its own built-in errors too! It would seem that in javascript though, popping from an empty array doesn't throw an error. It only returns an undefined. I would say though to do as you see fit, and if you insist on using an error, you could probably be justified with a nice `RangeError`